lxd/storage_volumes.go

```diff
@@ -895,6 +895,26 @@ func storagePoolVolumeTypeDelete(d *Daemon, r *http.Request, volumeTypeName stri
 		return SmartError(err)
 	}
 
+	// Delete storage volume snapshots
+	if volumeType == storagePoolVolumeTypeCustom {
```

Comment:
Depending on the type of storage backend you might have to do it the other way around and delete snapshots before you delete the actual volume.

Reply:
Are you sure? AFAICS snapshots are not tied to volumes.